lib/users_repository.rb

```diff
@@ -0,0 +1,70 @@
+require_relative 'users'
+
+class UsersRepository
+
+  # Selecting all records
+  # No arguments
+  def all
+    users = []
+    sql = 'SELECT id, username, email_address, password FROM users;'
+    result_set = DatabaseConnection.exec_params(sql, [])
+
+    result_set.each do |record|
+      user = Users.new
+
+      user.id = record['id']
+      user.username = record['username']
+      user.email_address = record['email_address']
+      user.password = record['password']
+
```

Comment:
Consider extracting lines 13-18 into a private method as the block is used for other methods in the file